core/container/src/main/java/org/wildfly/swarm/Swarm.java

```diff
@@ -464,7 +464,13 @@ private static boolean isFatJar() throws IOException {
 
     private void loadStageConfiguration(URL url) {
         List<ProjectStage> projectStages = new ProjectStageFactory().loadStages(url);
-        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE, "default");
+        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE);
```

Comment:
Also the suggested approach makes unnecessary calls if the property value exists, for example, so it should use orElseGet instead if converted